diff --git a/src/App.tsx b/src/App.tsx
index d697d1c..2fd4976 100644
--- a/src/App.tsx
+++ b/src/App.tsx
@@ -1,11 +1,14 @@
 import {NavigationContainer} from '@react-navigation/native';
+import {UserContextProvider} from 'contexts/UserContext';
 import {RootStackNav} from 'navigation';
 import React from 'react';
 
 export const App = () => {
   return (
-    <NavigationContainer>
-      <RootStackNav />
-    </NavigationContainer>
+    <UserContextProvider>
+      <NavigationContainer>
+        <RootStackNav />
+      </NavigationContainer>
+    </UserContextProvider>
   );
 };
diff --git a/src/navigation/UnauthedStackNav.tsx b/src/navigation/UnauthedStackNav.tsx
index ca652aa..e230631 100644
--- a/src/navigation/UnauthedStackNav.tsx
+++ b/src/navigation/UnauthedStackNav.tsx
@@ -2,7 +2,7 @@ import {useNavigation} from '@react-navigation/native';
 import {createStackNavigator} from '@react-navigation/stack';
 import React from 'react';
 import {Button} from 'react-native-elements';
-import {Welcome, Instructions, Login, TodoBoard} from 'screens';
+import {Welcome, Instructions, Login} from 'screens';
 
 const HeaderRight: React.FC = () => {
   const navigation = useNavigation();
@@ -10,7 +10,7 @@ const HeaderRight: React.FC = () => {
 };
 
 const nav = createStackNavigator();
-export const RootStackNav: React.FC = () => {
+export const UnauthedStackNav: React.FC = () => {
   return (
     <nav.Navigator
       initialRouteName={Welcome.name}
@@ -32,13 +32,6 @@ export const RootStackNav: React.FC = () => {
           headerRight: undefined,
         }}
       />
-      <nav.Screen
-        name="TodoBoard"
-        component={TodoBoard}
-        options={{
-          headerTitle: 'Todo',
-        }}
-      />
       <nav.Screen name="Instructions" component={Instructions} />
     </nav.Navigator>
   );
diff --git a/src/navigation/index.ts b/src/navigation/index.ts
index 7e4db67..9f57181 100644
--- a/src/navigation/index.ts
+++ b/src/navigation/index.ts
@@ -1 +1,3 @@
 export * from './RootStackNav';
+export * from './UnauthedStackNav';
+export * from './AuthedStackNav';
diff --git a/src/screens/auth/Login.tsx b/src/screens/auth/Login.tsx
index 7dd7160..bfae7c2 100644
--- a/src/screens/auth/Login.tsx
+++ b/src/screens/auth/Login.tsx
@@ -1,14 +1,19 @@
-import {useNavigation} from '@react-navigation/native';
-import React from 'react';
+import {useUserContext} from 'contexts/UserContext';
+import React, {useCallback} from 'react';
 import {StyleSheet, View} from 'react-native';
 import {Text, Button} from 'react-native-elements';
 
 export const Login: React.FC = () => {
-  const navigation = useNavigation();
+  const userContext = useUserContext();
+
+  const login = useCallback(() => {
+    userContext.login('user', 'password');
+  }, [userContext]);
+
   return (
     <View style={styles.container}>
       <Text h1>ログイン</Text>
-      <Button onPress={() => navigation.navigate('TodoBoard')} title="ログイン" />
+      <Button onPress={login} title="ログイン" />
     </View>
   );
 };
diff --git a/src/screens/todo/TodoBoard.tsx b/src/screens/todo/TodoBoard.tsx
index 7555435..caacb08 100644
--- a/src/screens/todo/TodoBoard.tsx
+++ b/src/screens/todo/TodoBoard.tsx
@@ -1,11 +1,17 @@
-import React from 'react';
+import {useUserContext} from 'contexts/UserContext';
+import React, {useCallback} from 'react';
 import {StyleSheet, View} from 'react-native';
-import {Text} from 'react-native-elements';
+import {Text, Button} from 'react-native-elements';
 
 export const TodoBoard: React.FC = () => {
+  const userContext = useUserContext();
+
+  const onLogout = useCallback(() => userContext.logout(), [userContext]);
+
   return (
     <View style={styles.container}>
       <Text h1>Todo</Text>
+      <Button type="clear" onPress={onLogout} title="ログアウト" />
     </View>
   );
 };
