diff --git a/src/navigation/AuthedStackNav.tsx b/src/navigation/AuthedStackNav.tsx
index 1d06db8..510f506 100644
--- a/src/navigation/AuthedStackNav.tsx
+++ b/src/navigation/AuthedStackNav.tsx
@@ -1,12 +1,12 @@
 import {createStackNavigator} from '@react-navigation/stack';
+import {MainTabNav} from 'navigation/MainTabNav';
 import React from 'react';
-import {TodoBoard} from 'screens';
 
 const nav = createStackNavigator();
 export const AuthedStackNav: React.FC = () => {
   return (
     <nav.Navigator screenOptions={{headerShown: false}}>
-      <nav.Screen name="Todo" component={TodoBoard} />
+      <nav.Screen name="Main" component={MainTabNav} />
     </nav.Navigator>
   );
 };
diff --git a/src/navigation/MainTabNav.tsx b/src/navigation/MainTabNav.tsx
new file mode 100644
index 0000000..bc381ac
--- /dev/null
+++ b/src/navigation/MainTabNav.tsx
@@ -0,0 +1,30 @@
+import {Ionicons} from '@expo/vector-icons';
+import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
+import React from 'react';
+
+import {SettingsStackNav} from './SettingsStackNav';
+import {TodoStackNav} from './TodoStackNav';
+
+const nav = createBottomTabNavigator();
+export const MainTabNav: React.FC = () => {
+  return (
+    <nav.Navigator initialRouteName="Todo">
+      <nav.Screen
+        name="Todo"
+        component={TodoStackNav}
+        options={{
+          title: 'ホーム',
+          tabBarIcon: ({color}) => <Ionicons name="md-home" size={30} color={color} />,
+        }}
+      />
+      <nav.Screen
+        name="Settings"
+        component={SettingsStackNav}
+        options={{
+          title: '設定',
+          tabBarIcon: ({color}) => <Ionicons name="md-settings" size={30} color={color} />,
+        }}
+      />
+    </nav.Navigator>
+  );
+};
diff --git a/src/navigation/SettingsStackNav.tsx b/src/navigation/SettingsStackNav.tsx
new file mode 100644
index 0000000..fe7dcb7
--- /dev/null
+++ b/src/navigation/SettingsStackNav.tsx
@@ -0,0 +1,10 @@
+import {createStackNavigator} from '@react-navigation/stack';
+import React from 'react';
+import {UserSetting} from 'screens';
+
+const nav = createStackNavigator();
+export const SettingsStackNav: React.FC = () => (
+  <nav.Navigator screenOptions={{headerShown: false}} initialRouteName="UserSetting">
+    <nav.Screen name="UserSetting" component={UserSetting} />
+  </nav.Navigator>
+);
diff --git a/src/navigation/TodoStackNav.tsx b/src/navigation/TodoStackNav.tsx
new file mode 100644
index 0000000..843a098
--- /dev/null
+++ b/src/navigation/TodoStackNav.tsx
@@ -0,0 +1,16 @@
+import {createStackNavigator} from '@react-navigation/stack';
+import React from 'react';
+import {TodoBoard} from 'screens';
+
+const nav = createStackNavigator();
+export const TodoStackNav: React.FC = () => (
+  <nav.Navigator initialRouteName="TodoBoard">
+    <nav.Screen
+      name="TodoBoard"
+      component={TodoBoard}
+      options={{
+        headerTitle: 'Todoアプリ',
+      }}
+    />
+  </nav.Navigator>
+);
diff --git a/src/navigation/index.ts b/src/navigation/index.ts
index 9f57181..739c056 100644
--- a/src/navigation/index.ts
+++ b/src/navigation/index.ts
@@ -1,3 +1,6 @@
 export * from './RootStackNav';
 export * from './UnauthedStackNav';
 export * from './AuthedStackNav';
+export * from './TodoStackNav';
+export * from './SettingsStackNav';
+export * from './MainTabNav';
diff --git a/src/screens/index.ts b/src/screens/index.ts
index 76fb4f8..a93d979 100644
--- a/src/screens/index.ts
+++ b/src/screens/index.ts
@@ -1,4 +1,5 @@
 export * from './auth';
 export * from './home';
 export * from './instructions';
+export * from './settings';
 export * from './todo';
diff --git a/src/screens/settings/UserSetting.tsx b/src/screens/settings/UserSetting.tsx
new file mode 100644
index 0000000..ea142ee
--- /dev/null
+++ b/src/screens/settings/UserSetting.tsx
@@ -0,0 +1,24 @@
+import {useUserContext} from 'contexts/UserContext';
+import React, {useCallback} from 'react';
+import {StyleSheet, View} from 'react-native';
+import {Button} from 'react-native-elements';
+
+export const UserSetting: React.FC = () => {
+  const userContext = useUserContext();
+
+  const onLogout = useCallback(() => userContext.logout(), [userContext]);
+
+  return (
+    <View style={styles.container}>
+      <Button type="clear" onPress={onLogout} title="ログアウト" />
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+});
diff --git a/src/screens/settings/index.ts b/src/screens/settings/index.ts
new file mode 100644
index 0000000..942051f
--- /dev/null
+++ b/src/screens/settings/index.ts
@@ -0,0 +1 @@
+export * from './UserSetting';
diff --git a/src/screens/todo/TodoBoard.tsx b/src/screens/todo/TodoBoard.tsx
index caacb08..7555435 100644
--- a/src/screens/todo/TodoBoard.tsx
+++ b/src/screens/todo/TodoBoard.tsx
@@ -1,17 +1,11 @@
-import {useUserContext} from 'contexts/UserContext';
-import React, {useCallback} from 'react';
+import React from 'react';
 import {StyleSheet, View} from 'react-native';
-import {Text, Button} from 'react-native-elements';
+import {Text} from 'react-native-elements';
 
 export const TodoBoard: React.FC = () => {
-  const userContext = useUserContext();
-
-  const onLogout = useCallback(() => userContext.logout(), [userContext]);
-
   return (
     <View style={styles.container}>
       <Text h1>Todo</Text>
-      <Button type="clear" onPress={onLogout} title="ログアウト" />
     </View>
   );
 };
