diff --git a/package.json b/package.json
index e995ccb..3481915 100644
--- a/package.json
+++ b/package.json
@@ -22,6 +22,7 @@
     "@react-navigation/stack": "~6.2.0",
     "expo": "^45.0.0",
     "expo-splash-screen": "~0.15.1",
+    "formik": "^2.2.6",
     "react": "17.0.2",
     "react-dom": "17.0.2",
     "react-native": "0.68.2",
@@ -30,7 +31,8 @@
     "react-native-reanimated": "~2.8.0",
     "react-native-safe-area-context": "3.3.2",
     "react-native-screens": "~3.11.1",
-    "react-native-web": "0.17.7"
+    "react-native-web": "0.17.7",
+    "yup": "^0.32.9"
   },
   "devDependencies": {
     "@babel/core": "^7.12.9",
diff --git a/src/screens/auth/Login.tsx b/src/screens/auth/Login.tsx
index 242be0d..6c8a320 100644
--- a/src/screens/auth/Login.tsx
+++ b/src/screens/auth/Login.tsx
@@ -1,28 +1,69 @@
 import {useUserContext} from 'contexts/UserContext';
+import {useFormik} from 'formik';
 import React, {useCallback} from 'react';
 import {StyleSheet, View} from 'react-native';
-import {Text, Button} from 'react-native-elements';
+import {Button, Input} from 'react-native-elements';
+import * as Yup from 'yup';
 
 export const Login: React.FC = () => {
   const userContext = useUserContext();
 
-  const login = useCallback(() => {
-    // 後で例外処理を追加するので、今はlintのエラーは無視してください。
-    userContext.login('user', 'password');
-  }, [userContext]);
+  const login = useCallback(
+    (values: {name: string; password: string}) => {
+      userContext.login(values.name, values.password).then(
+        () => {},
+        () => {},
+      );
+    },
+    [userContext],
+  );
+
+  const formik = useFormik({
+    initialValues: {name: '', password: ''},
+    validationSchema: Yup.object().shape({
+      name: Yup.string().required('名前を入力してください'),
+      password: Yup.string().required('パスワードを入力してください'),
+    }),
+    validateOnChange: false,
+    onSubmit: login,
+  });
 
   return (
-    <View style={styles.container}>
-      <Text h1>ログイン</Text>
-      <Button onPress={login} title="ログイン" />
+    <View style={styles.form}>
+      <Input
+        label="名前"
+        containerStyle={styles.input}
+        autoCapitalize="none"
+        errorMessage={formik.errors.name}
+        onChangeText={formik.handleChange('name')}
+        value={formik.values.name}
+      />
+      <Input
+        label="パスワード"
+        containerStyle={styles.input}
+        errorMessage={formik.errors.password}
+        secureTextEntry
+        onChangeText={formik.handleChange('password')}
+        value={formik.values.password}
+      />
+      <Button
+        disabled={formik.isSubmitting}
+        onPress={() => formik.handleSubmit()}
+        title="ログインする"
+        buttonStyle={styles.button}
+      />
     </View>
   );
 };
 
 const styles = StyleSheet.create({
-  container: {
+  form: {
     flex: 1,
     alignItems: 'center',
     justifyContent: 'center',
   },
+  input: {marginTop: 20, width: '80%'},
+  button: {
+    marginTop: 30,
+  },
 });
