diff --git a/src/screens/settings/UserSetting.tsx b/src/screens/settings/UserSetting.tsx
index ea142ee..d10954f 100644
--- a/src/screens/settings/UserSetting.tsx
+++ b/src/screens/settings/UserSetting.tsx
@@ -1,12 +1,21 @@
 import {useUserContext} from 'contexts/UserContext';
 import React, {useCallback} from 'react';
-import {StyleSheet, View} from 'react-native';
+import {Alert, StyleSheet, View} from 'react-native';
 import {Button} from 'react-native-elements';
 
 export const UserSetting: React.FC = () => {
   const userContext = useUserContext();
 
-  const onLogout = useCallback(() => userContext.logout(), [userContext]);
+  const onLogout = useCallback(() => {
+    Alert.alert('ログアウト確認', 'ログアウトします。\nよろしいですか？', [
+      {text: 'Cancel', style: 'cancel', onPress: () => {}},
+      {
+        text: 'OK',
+        style: 'destructive',
+        onPress: () => userContext.logout(),
+      },
+    ]);
+  }, [userContext]);
 
   return (
     <View style={styles.container}>
