diff --git a/src/components/basics/index.ts b/src/components/basics/index.ts
new file mode 100644
index 0000000..5638cb2
--- /dev/null
+++ b/src/components/basics/index.ts
@@ -0,0 +1 @@
+export * from './view';
diff --git a/src/components/basics/view/KeyboardView.tsx b/src/components/basics/view/KeyboardView.tsx
new file mode 100644
index 0000000..f0b8b0c
--- /dev/null
+++ b/src/components/basics/view/KeyboardView.tsx
@@ -0,0 +1,28 @@
+import React, {ComponentProps} from 'react';
+import {KeyboardAvoidingView, Platform, StyleSheet} from 'react-native';
+
+type KeyboardViewProps = ComponentProps<typeof KeyboardAvoidingView>;
+
+const defaultKeyboardAvoidingViewBehavior = Platform.select({
+  ios: 'padding',
+  android: undefined,
+} as const);
+
+export const KeyboardView: React.FC<KeyboardViewProps> = ({
+  children,
+  behavior = defaultKeyboardAvoidingViewBehavior,
+  style,
+  ...props
+}) => {
+  return (
+    <KeyboardAvoidingView behavior={behavior} style={StyleSheet.flatten([styles.container, style])} {...props}>
+      {children}
+    </KeyboardAvoidingView>
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+  },
+});
diff --git a/src/components/basics/view/index.ts b/src/components/basics/view/index.ts
new file mode 100644
index 0000000..d106650
--- /dev/null
+++ b/src/components/basics/view/index.ts
@@ -0,0 +1 @@
+export * from './KeyboardView';
diff --git a/src/screens/auth/Login.tsx b/src/screens/auth/Login.tsx
index 2751a2f..87dab3e 100644
--- a/src/screens/auth/Login.tsx
+++ b/src/screens/auth/Login.tsx
@@ -1,7 +1,8 @@
+import {KeyboardView} from 'components/basics';
 import {useUserContext} from 'contexts/UserContext';
 import {useFormik} from 'formik';
 import React, {useCallback} from 'react';
-import {KeyboardAvoidingView, Platform, StyleSheet, View} from 'react-native';
+import {StyleSheet, View} from 'react-native';
 import {Button, Input} from 'react-native-elements';
 import * as Yup from 'yup';
 
@@ -29,12 +30,7 @@ export const Login: React.FC = () => {
   });
 
   return (
-    <KeyboardAvoidingView
-      behavior={Platform.select({
-        ios: 'padding',
-        android: undefined,
-      } as const)}
-      style={styles.container}>
+    <KeyboardView>
       <View style={styles.form}>
         <Input
           label="名前"
@@ -59,14 +55,11 @@ export const Login: React.FC = () => {
           buttonStyle={styles.button}
         />
       </View>
-    </KeyboardAvoidingView>
+    </KeyboardView>
   );
 };
 
 const styles = StyleSheet.create({
-  container: {
-    flex: 1,
-  },
   form: {
     flex: 1,
     alignItems: 'center',
diff --git a/src/screens/todo/TodoForm.tsx b/src/screens/todo/TodoForm.tsx
index 2668e0e..570e67b 100644
--- a/src/screens/todo/TodoForm.tsx
+++ b/src/screens/todo/TodoForm.tsx
@@ -1,7 +1,8 @@
 import {useNavigation} from '@react-navigation/native';
+import {KeyboardView} from 'components/basics';
 import {useFormik} from 'formik';
 import React, {useCallback, useEffect} from 'react';
-import {Alert, KeyboardAvoidingView, Platform, StyleSheet, View} from 'react-native';
+import {Alert, StyleSheet, View} from 'react-native';
 import {Button, Input, Text} from 'react-native-elements';
 import {TodoService} from 'services';
 import * as Yup from 'yup';
@@ -46,12 +47,7 @@ export const TodoForm: React.FC = () => {
   }, [navigation, formik]);
 
   return (
-    <KeyboardAvoidingView
-      behavior={Platform.select({
-        ios: 'padding',
-        android: undefined,
-      } as const)}
-      style={styles.container}>
+    <KeyboardView>
       <View style={styles.form}>
         <Text h1>タスク登録</Text>
         <Input
@@ -69,14 +65,11 @@ export const TodoForm: React.FC = () => {
           buttonStyle={styles.addButton}
         />
       </View>
-    </KeyboardAvoidingView>
+    </KeyboardView>
   );
 };
 
 const styles = StyleSheet.create({
-  container: {
-    flex: 1,
-  },
   form: {
     flex: 1,
     alignSelf: 'stretch',
